serde-api/build.gradle

```diff
@@ -0,0 +1,85 @@
+plugins {
+    id 'java-library'
+    id 'signing'
+    id 'maven-publish'
+}
+
+tasks.register('sourceJar', Jar) {
+    archiveClassifier.set("sources")
+    from sourceSets.main.allJava
+}
+
+task javadocJar(type: Jar, dependsOn: javadoc) {
+    archiveClassifier.set("javadoc")
+    from javadoc.destinationDir
+}
+
+artifacts {
+    archives sourceJar, javadocJar
+}
+
+if (prod) {
+    signing {
+        sign(publishing.publications)
+    }
+}
+
+publishing {
+    if (prod) {
+        repositories {
+            maven {
+                url "https://s01.oss.sonatype.org/service/local/staging/deploy/maven2"
+                credentials {
+                    username sonatypeUsername
+                    password sonatypePassword
+                }
+            }
+        }
+    }
+
+    publications {
+        maven(MavenPublication) {
+            groupId = 'io.kafbat.ui'
+            artifactId = 'serde-api'
+            version = version
+
+            from components.java
+
+            artifact(sourceJar) {
+                classifier = 'sources'
+            }
+
+            artifact(javadocJar) {
+                classifier = 'javadoc'
+            }
```

Comment:
Is this needed? Do we publish our JAR to central, or who consumes us as JAR directly?

Reply:
yes we do
https://central.sonatype.com/artifact/io.kafbat.ui/serde-api

Reply:
ikr? one of the two hardest things in computer science

Reply:
So, we have consumers that use the API code somehow but not the UI? Any idea about what the use case is?

Reply:
@yeikel there are two things:
1) serde-api is the package to implement custom serdes, like https://github.com/kafbat/ui-serde-glue/
2) some people might use the app without a need for the frontend, to automate some stuff, like tests, that's a valid use case